data-refactoring-advisor/dra-frontend/src/Components/RefineCommunity-src-dest-color-working.js

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) 2023, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v1.0 as shown at https://oss.oracle.com/licenses/upl/ 
+
+import React, { useState, useEffect} from "react";
+import { Link } from "react-router-dom";
+import Select from "react-select";
+//import { Link, useNavigate } from "react-router-dom";
+import "../Css/styles.css";
+import { useLocation } from "react-router-dom";
+//import Pagination from "react-responsive-pagination";
+
+function RefineCommunity() {
+
+  const [values, setValues] = useState([]);
+  const [nodes, setNodes] = useState([]);
+
+  const [source, setSource] = useState("");
+  const [destination, setDestination] = useState("");
+  const [destinationArray, setDestinationArray] = useState([]);
+  const [destinations, setDestinations] = useState([]);
+  
+  const [sourceValue, setSourceValue] = useState("");
+  const [destinationValue, setDestinationValue] = useState("");
+
+  const [selectedCard, setSelectedCard] = "";
+  const location = useLocation();
+  let graphData =  JSON.parse(location.state.graphData);
+  
+  let sqlSetName = location.state.stsName;
+
+  useEffect(() => {
+    console.log("UseEffect Calling, Values :: "+ values)
+    setNodes(graphData.nodes);
+  },[]);
+  
+  function fetchDestinations() {
+    let arrayTemp = [];
+    const data = groupBy(graphData.nodes, 'color');
+    const destinations = [...new Set(Object.keys(data).map(objKey => objKey))];
+    destinations.map((item) => {
+      arrayTemp.push({"color":item})
+    })
```

Comment:
Please check if you can:
Add a "return" statement to this callback.
Callbacks of array methods should have return statements [javascript:S3796](https://sonarcloud.io/organizations/oracle/rules?open=javascript%3AS3796&rule_key=javascript%3AS3796)